pkg/recipes/terraform/config/providers/aws.go

```diff
@@ -139,10 +151,33 @@ func (p *awsProvider) generateProviderConfigMap(credentials *credentials.AWSCred
 		config[awsRegionParam] = region
 	}
 
-	if credentials != nil && credentials.AccessKeyCredential != nil &&
-		credentials.AccessKeyCredential.AccessKeyID != "" && credentials.AccessKeyCredential.SecretAccessKey != "" {
-		config[awsAccessKeyParam] = credentials.AccessKeyCredential.AccessKeyID
-		config[awsSecretKeyParam] = credentials.AccessKeyCredential.SecretAccessKey
+	if credentials != nil {
+		switch credentials.Kind {
+		case ucp_datamodel.AWSAccessKeyCredentialKind:
+			if credentials.AccessKeyCredential != nil &&
+				credentials.AccessKeyCredential.AccessKeyID != "" && credentials.AccessKeyCredential.SecretAccessKey != "" {
+				config[awsAccessKeyParam] = credentials.AccessKeyCredential.AccessKeyID
+				config[awsSecretKeyParam] = credentials.AccessKeyCredential.SecretAccessKey
+			}
+
+		case ucp_datamodel.AWSIRSACredentialKind:
+			// Radius requests will first be routed to STS endpoint,
+			// where it will be validated and then the request to the specific service (such as S3) will be made using
+			// the bearer token from the STS response.
+			// Based on the https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html,
+			// STS endpoint should be region based, and in the same region as
+			// Radius instance to minimize latency associated with STS call and thereby improve performance.
+			// We should provide the user with ability to configure the STS endpoint region.
+			// For now, we are using the global STS endpoint, which is the default.
+			// Ref. https://github.com/radius-project/radius/issues/7747
+			if credentials.IRSACredential != nil && credentials.IRSACredential.RoleARN != "" {
+				config[awsIRSAProvider] = map[string]any{
+					awsRoleARN:  credentials.IRSACredential.RoleARN,
+					sessionName: "radius-terraform-" + uuid.New().String(),
```

Comment:
What is the significance of this session name? Could you share some details on the motivation for the format here?

Reply:
I see, makes sense. We might want to consider operation id instead of uuid to map it to the corresponding request, doesn't need to happen in this PR because it will require changing API interface in a couple of places but something to look into as a follow up